src/resize.cc

```diff
@@ -694,8 +696,80 @@ class ResizeWorker : public NanAsyncWorker {
       image = gammaDecoded;
     }
 
+    // Apply normalization
+    if (baton->normalize) {
+      VipsInterpretation typeBeforeNormalize = image->Type;
+      if (typeBeforeNormalize == VIPS_INTERPRETATION_RGB) {
+        typeBeforeNormalize = VIPS_INTERPRETATION_sRGB;
+      }
+
+      // normalize the luminance band in LAB space:
+      VipsImage *lab;
+      if (vips_colourspace(image, &lab, VIPS_INTERPRETATION_LAB, NULL)) {
+        return Error();
+      }
+      vips_object_local(hook, lab);
+
+      VipsImage *luminance;
+      if (vips_extract_band(lab, &luminance, 0, "n", 1, NULL)) {
+        return Error();
+      }
+      vips_object_local(hook, luminance);
+
+      VipsImage *chroma;
+      if (vips_extract_band(lab, &chroma, 1, "n", 2, NULL)) {
+        return Error();
+      }
+      vips_object_local(hook, chroma);
+
+      VipsImage *stats;
+      if (vips_stats(luminance, &stats, NULL)) {
+        return Error();
+      }
+      vips_object_local(hook, stats);
+      double min = *VIPS_MATRIX(stats, 0, 0);
+      double max = *VIPS_MATRIX(stats, 1, 0);
+      double f = 100.0 / (max - min);
+      double a = -(min * f);
+
+      VipsImage *luminance100;
+      if (vips_linear1(luminance, &luminance100, f, a, NULL)) {
+        return Error();
+      }
+      vips_object_local(hook, luminance100);
+
+      VipsImage *normalized;
+      if (vips_bandjoin2(luminance100, chroma, &normalized, NULL)) {
+        return Error();
+      }
+      vips_object_local(hook, normalized);
+
+      VipsImage *origFormat;
+      if (vips_colourspace(normalized, &origFormat, typeBeforeNormalize, NULL)) {
+        return Error();
+      }
+      vips_object_local(hook, origFormat);
+
+      if (HasAlpha(image)) {
+        VipsImage *alpha;
+        if (vips_extract_band(image, &alpha, image->Bands - 1, "n", 1, NULL)) {
+          return Error();
+        }
+        vips_object_local(hook, alpha);
+
+        VipsImage *normalizedAlpha;
+        if (vips_bandjoin2(origFormat, alpha, &normalizedAlpha, NULL)) {
+          return Error();
+        }
+        vips_object_local(hook, normalizedAlpha);
+        image = normalizedAlpha;
+      } else {
+        image = origFormat;
+      }
+    }
+
     // Convert image to sRGB, if not already
-    if (image->Type != VIPS_INTERPRETATION_sRGB) {
+    if (image->Type != VIPS_INTERPRETATION_sRGB && image->Type != VIPS_INTERPRETATION_B_W) {
```

Comment:
good catch. Segfaults.
```
Thread 7 Crashed:
0   libwebp.5.dylib               	0x00000001042495af ImportYUVAFromRGBA + 1758
1   libwebp.5.dylib               	0x0000000104248df5 Import + 342
2   libwebp.5.dylib               	0x0000000104248667 Encode + 306
3   libwebp.5.dylib               	0x000000010424852f WebPEncodeRGB + 27
4   libvips.40.dylib              	0x00000001031a384d vips__webp_write_file + 173 (vips2webp.c:80)
5   libvips.40.dylib              	0x00000001031a31cf vips_foreign_save_webp_file_build + 79 (webpsave.c:149)
6   libvips.40.dylib              	0x00000001031c5e3c vips_object_build + 28 (object.c:342)
7   libvips.40.dylib              	0x00000001031d1825 vips_cache_operation_buildp + 69 (cache.c:840)
8   libvips.40.dylib              	0x00000001031d666c vips_call_required_optional + 876 (operation.c:858)
9   libvips.40.dylib              	0x00000001031d7147 vips_call_by_name + 103 (operation.c:898)
10  libvips.40.dylib              	0x00000001031d7225 vips_call_split + 149 (operation.c:1002)
11  libvips.40.dylib              	0x0000000103190fe2 vips_webpsave + 162 (foreign.c:2205)
12  sharp.node                    	0x0000000100bf1752 ResizeWorker::Execute() + 12194 (resize.cc:926)
13  node                          	0x000000010045b25f worker + 90
14  node                          	0x0000000100451952 uv__thread_start + 25
15  libsystem_pthread.dylib       	0x00007fff88ded899 _pthread_body + 138
16  libsystem_pthread.dylib       	0x00007fff88ded72a _pthread_start + 137
17  libsystem_pthread.dylib       	0x00007fff88df1fc9 thread_start + 13
```

So what's the special thing about greyscale and webp?